Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticSystem.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Physics.Experimental
+{
+
+    /// <summary>
+    /// Properties of the extent in which a damped
+    /// harmonic oscillator is free to move.
+    /// </summary>
+    [Serializable]
+    public struct ElasticExtentProperties<T>
+    {
+        /// <value>
+        /// Represents the lower bound of the extent,
+        /// specified as the norm of the n-dimensional extent
+        /// </value>
+        [SerializeField]
+        public float minStretch;
+
+        /// <value>
+        /// Represents the upper bound of the extent,
+        /// specified as the norm of the n-dimensional extent
+        /// </value>
+        [SerializeField]
+        public float maxStretch;
+
+        /// <value>
+        /// Whether the system, when approaching the upper bound,
+        /// will treat the upper bound as a snap point.
+        /// </value>
+        [SerializeField]
+        public bool snapToMax;
```

Comment:
Why is there a separate field for snapToMax, instead of just having another value in snapPoints which is equal to the max?

Reply:
i.e. as we think about how to grow/extend this struct in the future, should someone be adding snapToMin? snapToMiddle? Or when does it make sense to do that over having other points within snapPoints?

Reply:
Good point.

Reply:
This is configured this way as the "snap force" is applied slightly differently when it is the end cap. The magnetism/snap force is only applied if the current value is inside the extent; otherwise, it is not applied and only the one-sided force is applied. Otherwise, we would be "doubling up" with both snapping force and end-cap force.

The developer is free to disable endpoint snapping and add their own snapping points onto the ends, too.

Reply:
Can you add this difference to the comments for SnapToEnd?

This is an interesting behavior difference which isn't obvious from the existing comments about this:

/// Whether the system, when approaching the upper bound,
/// will treat the end limits like snap points and magnetize to them.

(i.e. it seems like the behavior is the same as snap points)